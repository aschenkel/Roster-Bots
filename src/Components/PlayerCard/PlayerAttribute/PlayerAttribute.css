.atributes-text{
    color:gray;
    text-shadow: 0px 1px 0px rgba(255,255,255,.3), 0px -1px 0px rgba(0,0,0,.7);

}
.icons{
  margin: 0px 8px 0 0;
  color:gray
}

.badge{
  margin: 0 8px 0 3px
}