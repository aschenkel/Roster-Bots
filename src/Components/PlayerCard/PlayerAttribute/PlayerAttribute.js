import React from 'react';
import PropTypes from 'prop-types';
import './PlayerAttribute.css';
import {MdFitnessCenter,MdDirectionsRun} from 'react-icons/lib/md'
import AttributeBadge from '../AttributeBadge/AttributeBadge'

const PlayerAttribute = (props) => {
    let icon
    switch (props.attributeType) {
        case attributes.speed:
                icon = <MdDirectionsRun size={23} className="icons"/>
            break;
        case attributes.strenght:
                icon = <MdFitnessCenter size={23} className="icons"/>
            break;
        default:
            break; 
    }
    return (
        <div>
            <AttributeBadge className="badge" value={props.value}/>
            <p className="atributes-text">{props.attributeType}</p>
        </div>
    )
}

export const attributes = {
    speed : "speed",
    strenght : "strenght",
}

PlayerAttribute.propTypes = {
  attributeType: PropTypes.string,
  attributeValue: PropTypes.number
};

export default PlayerAttribute;