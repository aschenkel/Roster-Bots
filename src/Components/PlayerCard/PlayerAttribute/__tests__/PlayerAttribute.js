import React from 'react';
import PlayerAttribute, {attributes} from '../PlayerAttribute';
import renderer from 'react-test-renderer';

it('should render a player´s attribute (speed)', () => {
    const tree = renderer.create(
        <PlayerAttribute attributeType={attributes.speed} attributeValue={44}/>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render a player´s attribute (strenght)', () => {
    const tree = renderer.create(
        <PlayerAttribute attributeType={attributes.strenght} attributeValue={44}/>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});

