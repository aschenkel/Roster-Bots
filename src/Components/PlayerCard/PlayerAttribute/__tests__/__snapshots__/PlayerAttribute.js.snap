// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a player´s attribute (speed) 1`] = `
<div>
  <p
    className="badge"
  />
  <p
    className="atributes-text"
  >
    speed
  </p>
</div>
`;

exports[`should render a player´s attribute (strenght) 1`] = `
<div>
  <p
    className="badge"
  />
  <p
    className="atributes-text"
  >
    strenght
  </p>
</div>
`;
